src/signatures/winternitz.rs

```diff
@@ -64,7 +64,7 @@ pub fn public_key_for_digit(secret_key: &str, digit_index: u32) -> [u8; 20] {
 pub fn generate_public_key(secret_key: &str) -> PublicKey {
     let mut public_key_array = [[0u8; 20]; N as usize];
     for i in 0..N {
-        public_key_array[i as usize] = public_key_for_digit(secret_key, i);
+        public_key_array[i as usize] = public_key_for_digit(secret_key, i.try_into().unwrap());
```

Comment:
Can we cast `i` to u32 instead of doing `try_into().unwrap()`?
```suggestion
        public_key_array[i as usize] = public_key_for_digit(secret_key, i as u32);
```